homeassistant/components/elmax/alarm_control_panel.py

```diff
@@ -83,6 +83,11 @@ async def async_alarm_arm_away(self, code: str | None = None) -> None:
 
     async def async_alarm_disarm(self, code: str | None = None) -> None:
         """Send disarm command."""
+        # Elmax alarm panels do always require a code to be passed for disarm operations
+        if code is None or code == "":
+            raise HomeAssistantError(
+                "Disarm operation failed: please input the disarm code."
+            )
```

Comment:
Not sure but perhaps this should raise `ValueError` instead as it's missing input?
And the operation did not fail as it did not try, Maybe `Mandatory disarm code not provided` or something as such?

Reply:
Addressed in a02a63b9d183236470e1c995d31fa36e0eb2c2c7, thanks!

Reply:
Did you test it with `ValueError` so it provides good feedback?